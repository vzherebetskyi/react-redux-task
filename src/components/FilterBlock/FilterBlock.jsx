import React, { useState, useEffect } from 'react';

import './FilterBlock.scss';

const FilterBlock = () => {
    const [filtOpened, setFiltOpened] = useState();
    const [searchValue, setSearchValue] = useState('');

    useEffect(() => {
      checkIsFiltersPreferenceSet();
    }, []);

    const checkIsFiltersPreferenceSet = () => {
        try {
          const isFiltersOpenedSet = localStorage.getItem('isFiltersOpened');
          if (isFiltersOpenedSet) {
            const isFiltersOpened = isFiltersOpenedSet === 'true' ? true : false;
            setFiltOpened(isFiltersOpened);
            return;
          }
          localStorage.setItem('isFiltersOpened', 'false');
        } catch (e) {
          console.error(e);
        }
    };

    const handleOpenFilterBlock = () => {
        try {
          localStorage.setItem('isFiltersOpened', 'true');
          setFiltOpened(true);
        } catch (e) {
          console.error(e);
        }
    }

    const handleCloseFilterBlock = () => {
        try {
          localStorage.setItem('isFiltersOpened', 'false');
          setFiltOpened(false);
        } catch (e) {
          console.error(e);
        }
    }
    
    return (
      <div className='filter-wrapper'>
        {filtOpened && 
        <div>
            <div className="input-container">
              <input
                autoComplete='off'
                placeholder='Search articles'
                type="text"
                name='search-articles'
                value={searchValue}
                onChange={e => setSearchValue(e.target.value)}
                // data-testid="base-input"
              />
              <p>* when input is empty the default search is applied (i.e. 'world news')</p>
            </div>
        </div>
        }
        {typeof filtOpened === 'boolean' && 
        <button className='filter-button' onClick={filtOpened ? () => handleCloseFilterBlock() : () => handleOpenFilterBlock()}>
            {filtOpened ? 'Hide filters' : 'Show filters'}
        </button>
        }
      </div>
  );
}

export default FilterBlock;
