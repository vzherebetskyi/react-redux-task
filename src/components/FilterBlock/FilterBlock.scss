.filter-wrapper {
  width: 100%;
  min-height: 20px;
  background: lightcoral;
  position: fixed;
  top: 0;
  padding: 5px;
  z-index: 999;
}

.filter-button {
  padding: 4px 8px;
  background: none;
  border: 1px solid #000;

  &:hover {
    cursor: pointer;
  }
}

.input-container {
  width: 22rem;
  position: relative;
  margin: 5px 0;

  input {
    height: 1.2rem;
    width: 15rem;
    border: 1px solid #000;
    border-radius: 5px;
    padding: 4px;

    &:focus, :active {
      outline: none;
    }
  }

  p {
    margin: 0;
  }
}