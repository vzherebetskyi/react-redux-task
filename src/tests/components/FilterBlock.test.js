import React from 'react';
import renderer from 'react-test-renderer';

import FilterBlock from '../../components/FilterBlock/FilterBlock';

test('should render FilterBlock component correctly', () => {
  const elem = renderer.create(<FilterBlock />).toJSON();
  expect(elem).toMatchSnapshot();
});
