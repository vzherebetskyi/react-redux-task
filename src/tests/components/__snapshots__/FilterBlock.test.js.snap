// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FilterBlock component correctly 1`] = `
<div
  className="filter-wrapper "
/>
`;
